src/Symfony/Component/Translation/Dumper/FileDumper.php

```diff
@@ -63,26 +63,34 @@ public function setBackup($backup)
      */
     public function dump(MessageCatalogue $messages, $options = array())
     {
-        if (!array_key_exists('path', $options)) {
-            throw new \InvalidArgumentException('The file dumper needs a path option.');
+        if (!array_key_exists('path', $options) && !array_key_exists('format_callback', $options)) {
```

Comment:
the name format_callback looks weird to me. This callback is not responsible for any formatting, but for saving